apm-agent-core/src/test/java/co/elastic/apm/agent/configuration/source/ConfigSourcesTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.configuration.source;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.stagemonitor.configuration.source.ConfigurationSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ConfigSourcesTest {
+
+    @Test
+    void loadFromClasspath() throws IOException {
+        ConfigurationSource source = ConfigSources.fromClasspath("test.elasticapm.properties", ClassLoader.getSystemClassLoader());
+        assertThat(source).isNotNull();
+
+        assertThat(source.getName()).isEqualTo("classpath:test.elasticapm.properties");
+        assertThat(source.getValue("application_packages")).isEqualTo("co.elastic.apm");
+        assertThat(source.getValue("log_level")).isEqualTo("DEBUG");
+
+        // should be no-op
+        source.reload();
```

Comment:
this doesn't show if it's not a no-op but doesn't throw an exception. Maybe change a value before the reload, then check that the reload didn't re-set the value back?

Reply:
Maybe it might be better to just check the implementation type (here it should be a `SimpleSource`), and not a subclass of it.